test/functional/wallet_upgrade.py

```diff
@@ -93,30 +121,27 @@ def run_test(self):
         self.stop_node(0)
         self.start_node(0, ["-upgradewallet"])
 
-        self.log.info("Checking old keys existence in the upgraded wallet..")
-        # Now check that all of the pre upgrade addresses are still in the wallet
-        for addr in addrs:
-            vaddr = self.nodes[0].getaddressinfo(addr)
-            assert_equal(vaddr['ismine'], True)
-            assert_equal('hdseedid' in vaddr, False)
-
-        self.log.info("All pre-upgrade keys found in the wallet :)")
+        # Now check if the upgrade went fine
+        self.check_keys(addrs)
+        self.log.info("New HD addresses created successfully")
 
-        # Use all of the keys in the pre-HD keypool
-        for _ in range(0, 60):
-            self.nodes[0].getnewaddress()
+        # Now test the upgrade at runtime using the JSON-RPC upgradewallet command
+        self.log.info("## Testing the runtime upgrade now, stopping the node..")
```

Comment:
nit: to avoid confusion, i think this should be changed to something like
```python
self.log.info("## Testing the upgrade via RPC now, stopping the node...")
```

rationale: "runtime" (or "at runtime") is typically used to describe something that happens automatically when the wallet is started. That would be a good term to use for the `-upgradewallet` startup flag, but not for the `upgradewallet` RPC command, which requires post-startup action.

Reply:
"at startup" is something that happens automatically when the wallet is started, "at runtime" denotes an action that is performed when the software is running at any time (the "runtime" concept applies to actions that occurs during the software execution).

Side from that, agree that we can just put "upgrade via RPC now" and done.